ambry-server/src/test/java/com.github.ambry.server/Http2ResponseHandler.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.github.ambry.server;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http2.HttpConversionUtil;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Process {@link io.netty.handler.codec.http.FullHttpResponse} translated from HTTP/2 frames
+ */
+@ChannelHandler.Sharable
+public class Http2ResponseHandler extends SimpleChannelInboundHandler<FullHttpResponse> {
+
+  protected final Logger logger = LoggerFactory.getLogger(getClass());
+  private StreamResult streamResult;
+
+  public Http2ResponseHandler() {
+
+  }
+
+  /**
+   * Track the {@link ChannelFuture} and {@link ChannelPromise}
+   *
+   * @param writeFuture A future that represent the request write operation
+   * @param promise The promise object that will be used to wait/notify events
+   * @return The previous object associated with {@code streamId}
+   * @see Http2ResponseHandler#awaitResponses(long, TimeUnit)
+   */
+  public void put(ChannelFuture writeFuture, ChannelPromise promise) {
+    streamResult = new StreamResult(writeFuture, promise);
```

Comment:
does this currently support a single stream per connection at a time? I guess for the integration tests, this should suffice for now.

Reply:
Yes, one stream per connection at a time. For test only now. Will improve it once we get enough client knowledge/experience.